Tests/Document/JSONObjectTests.swift

```diff
@@ -155,4 +155,31 @@ class JSONObjectTests: XCTestCase {
             XCTAssertEqual(idOfCompD, "d-2")
         }
     }
+
+    struct JsonObejctTestType: YorkieJSONObjectable {
+        var id: Int64 = 100
+        var type: String = "struct"
+        var serial: Int32 = 1234
+        var array: [JsonArrayTestType] = [JsonArrayTestType()]
```

Comment:
Arrays including types confirming YorkieJSONObjectable can be easily injected into the document.